framework/src/meshmodifiers/MeshExtruder.C

```diff
@@ -64,18 +73,18 @@ MeshExtruder::modify()
 
   _mesh_ptr->getMesh().clear();
 
-  QueryElemSubdomainID elem_subdomain_id(getParam<std::vector<SubdomainID> >("existing_subdomains"),
-                                         getParam<std::vector<unsigned int> >("layers"),
-                                         getParam<std::vector<unsigned int> >("new_ids"),
-                                         _num_layers);
+  QueryElemSubdomainID * elem_subdomain_id = NULL;
```

Comment:
Let's avoid `NULL` in new code. It can cause problems in certain C++11 situations, use `nullptr`.